frontend/src/utils/index.js

```diff
@@ -22,6 +28,65 @@ export const slugifiedNameFromFiles = files => {
   return slugify(projectWithExt.split('.')[0])
 }
 
+/**
+ * find readme file in a repo under a path(default root `/`)
+ * @param {Array<object>} repoFiles list of file names to search
+ * @param {string=} path a path(gitea path) to search into, if not passed searches repo path.
+ * @returns {string} the readme file name if found and an empty string if no readme file were found.
+ */
+export const findReadme = (repoFiles, path) => {
+  // TODO: implement path search.
+  /**
+   * @type {string[]}
+   */
+  const filesNames = repoFiles.map(f => f.name)
+  const isMatch = matcher('readme?(.markdown|.mdown|.mkdn|.md|.rst)', {
+    nocase: true,
+  })
+  // Find the first matching file index
+  const fileIndex = filesNames.map(f => isMatch(f)).indexOf(true)
+
+  return fileIndex !== -1 ? filesNames[fileIndex] : ''
+}
+
+/**
+ * convert readme to html and convert urls to absolute urls.
+ * @param {string} repo
+ * @param {string} readmeFile
+ * @returns {Promise<string>} the content of readme file as html
+ */
+export const renderReadme = async (repo, readmeFile) => {
+  // TODO handle `rst` case.
+  const readmeContent = await getFileRawContent(repo, readmeFile)
+  const readmeAsHtml = await renderMarkdown(readmeContent)
+
+  // Replace relative urls with absolute ones for `img` and `a` tags.
+  const $ = cheerio.load(readmeAsHtml)
+  $('img').each((_, elem) => {
+    const img = $(elem)
+    const src = img.attr('src')
+    // Used for styling images inside readme, in _app.css
+    img.attr('class', 'readmeImg')
+
+    if (src.startsWith('/')) {
+      const rawUrl = `${giteaUrl}/${repo}/raw${src}`
+      img.attr('src', rawUrl)
+    }
+  })
+
+  $('a').each((_, elem) => {
+    const a = $(elem)
+    const href = a.attr('href')
+    if (href.startsWith('/')) {
+      const rawUrl = `${giteaUrl}/${repo}/src${href}`
+      console.log(rawUrl)
+      a.attr('href', rawUrl)
+    }
+  })
+
+  return $.html()
+}
```

Comment:
I thought we would use the Gitea readme renderer.

Reply:
Yes, I did so. This part only fixes the URLs.

Reply:
We should make an issue to do this in Gitea itself. Also, there's a `console.log` in there.

Reply:
Made an issue: https://github.com/kitspace/kitspace-v2/issues/128